sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -568,17 +572,28 @@ case class MakeYMInterval(years: Expression, months: Expression)
   override def dataType: DataType = YearMonthIntervalType()
 
   override def nullSafeEval(year: Any, month: Any): Any = {
-    Math.toIntExact(Math.addExact(month.asInstanceOf[Number].longValue(),
-      Math.multiplyExact(year.asInstanceOf[Number].longValue(), MONTHS_PER_YEAR)))
+    try {
+      Math.toIntExact(
+        Math.addExact(month.asInstanceOf[Int],
+          Math.multiplyExact(year.asInstanceOf[Int], MONTHS_PER_YEAR)))
+    } catch {
```

Comment:
shall we add `IntervalUtils.makeYearMonthInterval` so that we duplicate less code between interpreted and codegen?

Reply:
Off course we should deduplicate it, but I didn't succeed with it: https://github.com/apache/spark/pull/48773#discussion_r1831606335.
Max suggested to use StaticInvoke, I'll try it in separate PR.

Reply:
there is already `makeDayTimeInterval` and we can just follow it. Can you try it again and create a PR? We can help to check the code together.

Reply:
Ok, I'll do it a bit later.